lib/testutils/src/lib.rs

```diff
@@ -450,3 +462,22 @@ pub fn assert_rebased(
         panic!("expected rebased commit: {rebased:?}");
     }
 }
+
+pub fn assert_abandoned(
```

Comment:
It seems that this ends up being used only in `test_rewrite.rs`. I would prefer to define it in that file (and in the commit where it's first used) until we find a use for it in other files.